meta-mender-tegra/recipes-mender/tegra-state-scripts/files/redundant-boot-install-script-uboot

```diff
@@ -46,12 +46,16 @@ fi
 mount --bind /sys "${mnt}/sys"
 mount --bind /proc "${mnt}/proc"
 mount --bind /dev "${mnt}/dev"
+
+# Mount an overlay for /var/lib so the /var/lib/nv_boot_control.conf file becomes writable
+mount -t tmpfs tmpfs ${mnt}/var/volatile
+mount-copybind ${mnt}/var/volatile/lib ${mnt}/var/lib/
+
 # Run the update engine in the context of the just-installed rootfs
 # to ensure that the TNSPEC in the config file it uses matches the
 # TNSPEC in the update payload. But first we have to set up the
 # configuration file with the TNSPEC.
 if [ -L "${mnt}/etc/nv_boot_control.conf" -a -x "${mnt}/usr/bin/setup-nv-boot-control" ]; then
-    rm "${mnt}/etc/nv_boot_control.conf"
```

Comment:
@nielsavonds @madisongh this seems like the right change to make but I'm wondering if there are any side-effects and trying to understand/remember the reason it was done this way initially in https://github.com/mendersoftware/meta-mender-community/commit/b09345a6286d545ea0292e168184724f29418446.  
With the `rm` line included as it was previously I think we are going to be replacing the symlink originally created at https://github.com/OE4T/meta-tegra/blob/dunfell-l4t-r32.4.3/recipes-bsp/tegra-binaries/tegra-nv-boot-control-config_32.4.3.bb#L23 with a file in the root filesystem after the first mender update.  It seems like it would be better (and would play nicer with delta updates) to always keep the symlink around and always write it on boot as we would do after these changes.
The best long term strategy I suppose is to [get rid of the nv_boot_control.conf headaches completely](https://github.com/OE4T/tegra-boot-tools#differences-from-nv_update_engine) with https://github.com/OE4T/tegra-boot-tools.  For dunfell though and to play nicely with delta updates I'm wondering if we should rework to use the scheme here for all cboot and uboot MACHINEs, or if there's some pitfall with this approach I don't know about.

Reply:
> It seems like it would be better (and would play nicer with delta updates) to always keep the symlink around and always write it on boot as we would do after these changes.

The reason why the script removes the symlink and writes a file is that the format of the configuration file has changed from version to version of the BSP.  Overwriting the existing configuration file with the new version could result in breakage should there be a rollback (without restoring the original config file) or any other boot slot switchover where the config file format is incompatible with the version of the tools in the rootfs.

Reply:
> Overwriting the existing configuration file with the new version could result in breakage should there be a rollback (without restoring the original config file) or any other boot slot switchover where the config file format is incompatible with the version of the tools in the rootfs.

Wouldn't that scenario be handled by https://github.com/OE4T/meta-tegra/blob/dunfell-l4t-r32.4.3/recipes-bsp/tools/setup-nv-boot-control/setup-nv-boot-control.sh.in#L19 on the boot to either the new or rollback rootfs?  As long as the symlink exists doesn't the logic in `setup-nv-boot-control` ensure that the content of the `nv_boot_control.template` in the new or rollback rootfs is used to format `nv_boot_control.conf` correctly for whatever the corresponding version of the BSP expects?

Reply:
No, if the symlink points to an existing file, the script exits early (see line 6).

Reply:
Thanks @madisongh and summarizing our slack discussion:
* `/var/lib` will be redirected to the `/data` partition if using a fstab setup like the one at https://github.com/OE4T/tegra-demo-distro/blob/e46346f6b331a76d0d9b28210e258be27f0b59a2/layers/meta-tegrademo/recipes-core/base-files/base-files/tegrademo-mender/fstab#L6, otherwise it will be redirected to `/var/volatile` by https://github.com/openembedded/openembedded-core/blob/master/meta/recipes-core/volatile-binds/volatile-binds.bb#L19
* To handle in a common way regardless of `/var/lib` mapping to volatile or non-volatile and to support read only rootfs and delta updates, we could symlink `/etc/nv_boot_control.conf` to `/run/tegra-nv-boot-control/nv_boot-control.conf` in a bbappend for https://github.com/OE4T/meta-tegra/blob/dunfell-l4t-r32.4.3/recipes-bsp/tegra-binaries/tegra-nv-boot-control-config_32.4.3.bb#L23 and then in https://github.com/OE4T/meta-tegra/blob/dunfell-l4t-r32.4.3/recipes-bsp/tools/setup-nv-boot-control/setup-nv-boot-control.sh.in#L6 we could `mkdir -p` the `readlink` result for the symlink target of `"${sysconfdir}/nv_boot_control.conf"`

Reply:
Follow-up below at https://github.com/mendersoftware/meta-mender-community/pull/208#issuecomment-751517083